### ISSUE TITLE
Adding new tables to group0 is awkward due to how snapshot transfer works

### ISSUE THREAD

**Issue by piodul** [scylladb/scylladb#19990]
This is how, currently, the `raft_pull_snapshot` RPC looks like:

```c++
struct raft_snapshot {
    utils::chunked_vector<canonical_mutation> mutations;
};

struct raft_snapshot_pull_params {
    std::vector<table_id> tables;
};

verb [[cancellable]] raft_pull_snapshot (raft::server_id dst_id, service::raft_snapshot_pull_params) -> service::raft_snapshot;
```

When a follower requests a snapshot from the leader, it sends a list of table IDs. The leader returns mutations representing the contents of the requested tables in response.

Now, let's consider that we want to add a new table to group 0 (concrete example: [view_build_status](https://github.com/scylladb/scylladb/issues/15329)). At this point, we need to do it in a way which takes into account rolling upgrades from versions which did not have this table.

A naive and wrong idea which could come to mind would be to conditionally add the ID of the new table to the `tables` field, depending on whether a _cluster feature_ is enabled on the requesting node (we would introduce a new cluster feature for that). However, this is wrong:

- We would most likely also want to use the new cluster feature to decide whether it's ok to generate group 0 commands that write to the new table. If so, it might happen that node A (leader) enables the feature and applies some commands already, then node B which has not enabled the feature requests the snapshot transfer and misses the updates to the new table
- This doesn't work at all during node join. When a node joins, it pulls a raft snapshot and uses the newly pulled state to determine which features need to be enabled. Such nodes will never decide to pull the new table.

There is a way to make snapshot transfer and cluster features work together for this purpose, but it requires a slight change in how snapshot transfer works: instead letting the follower choose which tables to fetch, it should be the _leader_ who chooses which tables to send. Because information about enabled cluster features is now included in the group0 state, there is no risk of choosing to send too little data.

In a nutshell, I propose deprecating the `raft_snapshot_pull_params::params` field and using cluster features on the RPC receiver side to choose the tables to send to.

**Comment by piodul** [comment]
cc: @kbr-scylla @gleb-cloudius @kostja @avikivity

**Comment by kostja** [comment]
I think @nuivall is addressing this at the moment.

**Comment by dani-tweig** [comment]
@mlitvk , @piodul , can we move this to the backlog or close it?
